flink-connectors/flink-connector-dynamodb/src/main/java/org/apache/flink/streaming/connectors/dynamodb/util/AWSDynamoDbUtil.java

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
As an alternative we could let user to construct the client themselves and pass it via a provider:

`
interface DynamoDbClientProvider {
   DynamoDbAsyncClient getClient(); 
}
`

Then we do not have to create the wrapper util class and give full flexibility to the user on what configuration options to set.

The downside will be that we leak DynamoDB SDK interfaces. 

In the end I took same approach as we use in other places. But may be we should consider this option.

What do you think @CrynetLogistics ?

Reply:
I like your current approach more, like you said, we'd not leak any ddb sdk interfaces.

Plus, I'm not sure there is a way to pass `DynamoDbClientProvider` to the `DynamoDbSinkWriter` without first making it a field in `DynamoDbSink`. If I'm not wrong, I believe the `DynamoDbSink` will complain that `DynamoDbAsyncClient` in your `DynamoDbClientProvider` is not serialisable during runtime (it compiles ok).

Reply:
I tested with provider in the previous sink implementation. Marking the client field as transient on the Sink helps with the serialisation issue in runtime.

But I will stick with the current approach for consistency and other reasons you mentioned

Reply:
Ah good to know. Thanks, agreed.